### PR TITLE
Story WORK-63 | Create Primal CLI

### PR DESCRIPTION
**Pull request by Rmgreco**
**Type**: Story

**Section**: cmd folder

**Summary**: 
- Created Primal CLI with develop and build modes/commands
- Integrated CLI with Primal's new architecture

**Pay attention to**:
The usability of the CLI. See ways that prompted messages and command usage can be improved

### CODE REVIEW

File: cmd/primal/cmd/utils.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+	"inspr.dev/primal/pkg/api"
+)
+
+var defaultPath string
+var inputPath string
+
+type TestYaml struct {
```
Comment (aluchianic) [inline cmd/primal/cmd/utils.go:15]: seems that this struct isn't in use, can we remove that ?

**Review by aluchianic — APPROVED** [review]

File: cmd/primal/cmd/build.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"inspr.dev/primal/pkg/platform"
+)
+
+// buildCmd implements primal build command
+var buildCmd = &cobra.Command{
+	Use:   "build [platform]",
+	Short: "Runs Primal in production mode",
+	Long: `Runs Primal in production mode, building and generating production-ready files.
+	The platform can be one of “electron”, “web”`,
+	Example: "primal build electron -f dir/config.yaml",
```
Comment (brenordr) [inline cmd/primal/cmd/build.go:19]: @franciscobonand should we add a default path?
For example if we don't set the -f we look for the current folder location

I've used most of the time like it, and can be convenient

File: cmd/primal/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+func initConfig() {
+	if cfgFile != "" {
+		viper.SetConfigFile(cfgFile)
+	} else {
+		home, err := os.UserHomeDir()
+		cobra.CheckErr(err)
+
+		viper.AddConfigPath(home)
+		viper.SetConfigType("yaml")
+		viper.SetConfigName(".primal")
```
Comment (brenordr) [inline cmd/primal/cmd/root.go:50]: What is the purpose of this config?

**Review by brenordr — COMMENTED** [review]
Looks good, I just have some questions

File: cmd/primal/cmd/utils.go (hunk 1 above)
Comment (franciscobonand) [reply cmd/primal/cmd/utils.go:15]: Yes we can

File: cmd/primal/cmd/root.go (hunk 3 above)
Comment (franciscobonand) [reply cmd/primal/cmd/root.go:50]: I think it was automatically generated by viper, can probably be removed

**Pull request merged**